topics/single-cell/tutorials/pseudobulk-analysis/tutorial.md

```diff
@@ -0,0 +1,465 @@
+> >    - **Filter by Expression Plot:** Illustrates the expression filtering applied to the data.
+> >      ![Filter by Expression Plot Example](../../images/pseudobulk-analysis/Decoupler_Filter_by_Expression_Plot.png)
+> >    - **Pseudobulk AnnData file:** An AnnData file that contains the aggregated pseudobulks.
+> >
+> > 3. The **pseudobulk count matrix** is the primary input required for analysis using **edgeR**, a tool designed for differential expression analysis.
+> >    The **Samples Metadata** is another file that will serve as an input for the edgeR tool.
+> >
+> {: .solution}
+>
+{: .question}
+
+## Sanitation Steps - Part 1
+
+The next steps will help you refine your data for easier handling. We will use standard galaxy tools, like: [Replace Text](https://toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1), [Remove columns](https://toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0) and [Text reformatting](https://toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1).
+
+> <hands-on-title>Replace Text and Remove Columns</hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `count_matrix` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"Find pattern"*: `[ --+*^]+`
+>            - *"Replace with:"*: `_`
+>
+> 2. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `samples_metadata` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"Find pattern"*: `[ --+*^]+`
+>            - *"Replace with:"*: `_`
+>
+> 3. {% tool [Remove columns](toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0) %} with the following parameters:
+>    - {% icon param-file %} *"Tabular file"*: `genes_metadata` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Select Columns"*:
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `start`
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `end`
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `width`
+>
+> 4. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `outfile` (output of **Replace Text: Sample Metadata Step** {% icon tool %})
```

Comment:
since step 4 works on the output of step 2, would it make sense to move it to step 3? finish processing of the sample metadata before moving on to the next file?